tools/serverpod_cli/lib/src/analyzer/models/definitions.dart

```diff
@@ -253,6 +261,35 @@ class ProtocolEnumValueDefinition {
   ProtocolEnumValueDefinition(this.name, [this.documentation]);
 }
 
+abstract class InheritanceDefinition {
+  final String className;
+
+  InheritanceDefinition(this.className);
+}
+
+class UnresolvedInheritanceDefinition extends InheritanceDefinition {
+  UnresolvedInheritanceDefinition(super.className);
+}
+
+class BaseClassInheritanceDefinition extends InheritanceDefinition {
+  final List<ClassDefinition?> subClasses;
+
+  BaseClassInheritanceDefinition(
+    super.className,
+    this.subClasses,
+  );
+}
+
+class ExtendsClassInheritanceDefinition extends InheritanceDefinition {
+  final List<SerializableModelFieldDefinition> baseClassFields;
+  final String baseClassPath;
+  ExtendsClassInheritanceDefinition(
+    super.className,
+    this.baseClassFields,
+    this.baseClassPath,
+  );
+}
```

Comment:
Suggestion/Discussion: With this definition of the class, we need to know up front what we want to retrieve for any use of it. How about we simply store the `ClassDefinition` and let the library generator pick out the fields that are required?

Reply:
Same for the `BaseClassInheritanceDefinition`.